src/Parser.ts

```diff
@@ -124,14 +124,19 @@ function stripRules(token: IToken, re: RegExp) {
 export interface IDictionary<T> {
   [s: string]: T;
 }
+export interface IParserOptions {
+  keepUpperRules: boolean;
+  debug: boolean;
+}
 
 const ignoreMissingRules = ['EOF'];
 
 export class Parser {
-  debug = false;
+  private readonly debug;
```

Comment:
```suggestion
  private readonly debug: boolean;
```

Reply:
Why is it readonly now?

Reply:
Because the debug option is provided during construction. But it is absolutely not required.